src/v/cluster/metadata_dissemination_service.cc

```diff
@@ -371,6 +371,10 @@ ss::future<> metadata_dissemination_service::dispatch_disseminate_leadership() {
 
 ss::future<> metadata_dissemination_service::update_leaders_with_health_report(
   cluster_health_report report) {
+    if (report.was_metadata_refreshed == refreshed_metadata::no) {
+        co_return;
+    }
```

Comment:
Not a hot path so feel free to punt, but maybe avoid scheduling this future by just returning at the call-site at L357? In which case, maybe we make this condition here a `vassert` so future writers aren't tempted to call this with cached/potentially-stale metadata